src/engine/server/sv_client.cpp

```diff
@@ -1402,7 +1402,7 @@ void SV_ExecuteClientMessage( client_t *cl, msg_t *msg )
 	// NOTE: when the client message is fux0red the acknowledgement numbers
 	// can be out of range, this could cause the server to send thousands of server
 	// commands which the server thinks are not yet acknowledged in SV_UpdateServerCommandsToClient
-	if ( cl->reliableAcknowledge < cl->reliableSequence - MAX_RELIABLE_COMMANDS )
```

Comment:
This patch is kinda poorly written - the subtraction is UB if they send a big negative number. How about `cl->reliableAcknowledge < 0 || cl->reliableAcknowledge < cl->reliableSequence - MAX_RELIABLE_COMMANDS || cl->reliableAcknowledge > cl->reliableSequence` (preventing negative ones as a bonus)

Reply:
Ok, replaced the patch with one doing what you suggest.

Reply:
LGTM